dags/remote_automated_ampliseq.py

```diff
@@ -0,0 +1,373 @@
+--outdir {REMOTE_PATH}/{study['study']}/out \
+-c {REMOTE_CONFIG}/ampliseq.config \
+-profile singularity
+echo "nextflow_done"
+exit
+    '''
+            script_path = os.path.join(study['path'], script_name)
+            with open(script_path, 'w') as file:
+                file.write(script_content)
+            return {"study": study, "script_name": script_name}
+
+        @dag.task(trigger_rule=TriggerRule.ALL_SUCCESS)
+        def compress_local_directory(study):
+            logging.info(f"Compressing local directory for study {study['study']}...")
+            tar_filename = f"{study['study']}_data.tar.gz"
+            local_tar_path = os.path.join(study['path'], tar_filename)
+            with tarfile.open(local_tar_path, "w:gz") as tar:
+                tar.add(study['path'], arcname=os.path.basename(study['path']))
+            return {"study": study, "tar_filename": tar_filename}
+
+        @dag.task(trigger_rule=TriggerRule.ALL_SUCCESS)
+        def transfer_compressed_directory(tar_info):
+            logging.info(f"Transferring compressed directory for study {tar_info['study']['study']}...")
+            study = tar_info["study"]
+            tar_filename = tar_info["tar_filename"]
+            local_tar_path = os.path.join(study['path'], tar_filename)
+            remote_tar_path = os.path.join(REMOTE_PATH, study['study'], tar_filename)
+            sftp_op = SFTPOperator(
+                task_id='transfer_compressed_directory_' + study['study'],
+                ssh_hook=SSHHook(
+                    remote_host=REMOTE_HOST_SFTP,
+                    username=USERNAME,
+                    key_file=KEY_FILE
+                ),
+                local_filepath=local_tar_path,
+                remote_filepath=remote_tar_path,
+                operation='put',
+                create_intermediate_dirs=True
+            )
+            sftp_op.execute(context={})
+
+        @dag.task(trigger_rule=TriggerRule.ALL_SUCCESS)
+        def extract_remote_directory(tar_info):
+            logging.info(f"Extracting remote directory for study {tar_info['study']['study']}...")
+            study = tar_info["study"]
+            tar_filename = tar_info["tar_filename"]
+            remote_tar_path = os.path.join(REMOTE_PATH, study['study'], tar_filename)
+            command = f'mkdir -p {REMOTE_PATH}/{study["study"]} && tar --strip-components=1 -xzf {remote_tar_path} -C {REMOTE_PATH}/{study["study"]} && rm {remote_tar_path}'
+            ssh_op = CustomSSHOperator(
+                task_id='extract_remote_directory_' + study['study'],
+                ssh_hook=SSHHook(
+                    remote_host=REMOTE_HOST_SSH,
+                    username=USERNAME,
+                    key_file=KEY_FILE
+                ),
+                command=command
+            )
+            ssh_op.execute(context={})
+
+        @dag.task(trigger_rule=TriggerRule.ALL_SUCCESS)
```

Comment:
id love to have added fetchngs as option to the ampliseq dag as well, rather than having to do this, but am ok leaving it out of scope for now.